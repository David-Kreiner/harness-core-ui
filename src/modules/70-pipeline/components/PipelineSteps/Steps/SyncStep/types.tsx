/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { AllowedTypes, MultiSelectOption } from '@harness/uicore'
import type { FormikProps } from 'formik'
import type { GetDataError } from 'restful-react'
import type { StepViewType } from '@pipeline/components/AbstractSteps/Step'
import type { Failure } from 'services/cd-ng'
import type { AllFailureStrategyConfig } from '../../AdvancedSteps/FailureStrategyPanel/utils'

export enum SYNC_RESOURCES {
  ALL = 'All',
  OUT_OF_SYNC = 'OutOfSync'
}

export enum POLICY_OPTIONS {
  FOREGROUND = 'foreground',
  BACKGROUND = 'background',
  ORPHAN = 'orphan'
}

export const policyOptions = [
  {
    label: 'Foreground',
    value: POLICY_OPTIONS.FOREGROUND
  },
  {
    label: 'Background',
    value: POLICY_OPTIONS.BACKGROUND
  },
  {
    label: 'Orphan',
    value: POLICY_OPTIONS.ORPHAN
  }
]

export interface applicationListItemInterface {
  applicationName: string
  agentId: string
}

export interface retryStrategyInterface {
  limit?: string
  baseBackoffDuration?: string
  increaseBackoffByFactor?: string
  maxBackoffDuration?: string
}

export interface syncOptionsInterface {
  skipSchemaValidation: boolean
  autoCreateNamespace: boolean
  pruneResourcesAtLast: boolean
  applyOutOfSyncOnly: boolean
  replaceResources: boolean
  prunePropagationPolicy: POLICY_OPTIONS
}

export interface SyncStepSpec {
  prune: boolean
  dryRun: boolean
  applyOnly: boolean
  forceApply: boolean
  applicationsList: applicationListItemInterface[] | string | MultiSelectOption[]
  retryStrategy: retryStrategyInterface
  retry?: boolean
  syncOptions: syncOptionsInterface
}

export interface SyncStepData {
  identifier: string
  name?: string
  type: string
  timeout?: string
  failureStrategies?: AllFailureStrategyConfig[]
  spec: SyncStepSpec
}

export interface SyncStepFormContentInterface {
  formik: FormikProps<SyncStepData>
  stepViewType: StepViewType
  allowableTypes: AllowedTypes
  projectsFetchError?: GetDataError<Failure | Error> | null
  projectMetadataFetchError?: GetDataError<Failure | Error> | null
  isNewStep?: boolean
  readonly?: boolean
}

export interface ApplicationItemInterface {
  item: MultiSelectOption
  disabled: boolean
  handleClick: () => void
}

export enum HealthStatus {
  Unknown = 'Unknown',
  Progressing = 'Progressing',
  Suspended = 'Suspended',
  Healthy = 'Healthy',
  Unhealthy = 'Unhealthy',
  Degraded = 'Degraded',
  Missing = 'Missing'
}

export enum SyncStatus {
  Unknown = 'Unknown',
  Synced = 'Synced',
  OutOfSync = 'OutOfSync'
}

export interface ApplicationFilters {
  page: number
  size: number
  search: string
  agents: string[]
  healthStatus: HealthStatus[]
  syncStatus: SyncStatus[]
  namespaces: string[]
  repo: string[]
}

export interface ApplicationFilterActions {
  goToPage(page: number): void
  search(term: string): void
  clearFilters(): void
  reset(): void
  healthStatus(status: HealthStatus[]): void
  syncStatus(status: SyncStatus[]): void
  agents(id: string[]): void
  namespaces(namespaces: string[]): void
  repo(repo: string[]): void
}
