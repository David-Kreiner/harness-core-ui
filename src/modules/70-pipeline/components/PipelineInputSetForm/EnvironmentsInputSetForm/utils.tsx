/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { get, isEmpty } from 'lodash-es'

import { RUNTIME_INPUT_VALUE, SelectOption } from '@harness/uicore'

import { DeploymentStageConfig } from 'services/cd-ng'
import { StageElementWrapperConfig } from 'services/pipeline-ng'

import { isValueRuntimeInput } from '@common/utils/utils'

import { StageType } from '@pipeline/utils/stageHelpers'
import { DeploymentStageElementConfig } from '@pipeline/utils/pipelineTypes'

export interface PropagateSelectOption extends SelectOption {
  infraLabel?: string
}

export function getPropagateStageOptions(
  stages: StageElementWrapperConfig[],
  stageIndex: number,
  templateTypes: {
    [key: string]: string
  }
): PropagateSelectOption[] {
  if (stages.length && stageIndex > 0) {
    const getStagesAllowedforPropagate = (stageItem: StageElementWrapperConfig): boolean => {
      if (stageItem.stage?.template) {
        const isAlreadyPropagatingValue = !!(stageItem?.stage?.template?.templateInputs?.spec as DeploymentStageConfig)
          ?.environment?.useFromStage?.stage
        const stageType = get(templateTypes, stageItem.stage.template.templateRef)
        return stageType === StageType.DEPLOY && !isAlreadyPropagatingValue
      } else {
        const isAlreadyPropagatingValue = !!(stageItem?.stage?.spec as DeploymentStageConfig)?.environment?.useFromStage
          ?.stage
        const stageType = stageItem?.stage?.type
        const isSingleEnvEmpty = isEmpty(
          (stageItem.stage as DeploymentStageElementConfig)?.spec?.environment?.environmentRef
        )
        return stageType === StageType.DEPLOY && !isSingleEnvEmpty && !isAlreadyPropagatingValue
      }
    }

    const stageWithEnvV2 = stages.slice(0, stageIndex).filter(getStagesAllowedforPropagate)
    return stageWithEnvV2.map(stageItem => {
      if (stageItem.stage?.template) {
        return {
          label: `Stage [${stageItem.stage?.name}] - Template [${stageItem.stage.template.templateRef}]`,
          value: stageItem.stage?.identifier as string,
          infraLabel: `Stage [${stageItem.stage?.name}] - Template [${stageItem.stage.template.templateRef}]`
        }
      } else {
        const singleEnvironmentRef = (stageItem.stage as DeploymentStageElementConfig)?.spec?.environment
          ?.environmentRef
        const environmentLabel = `Environment [${singleEnvironmentRef}]`

        const singleInfrastructureRef = isValueRuntimeInput(
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
          (stageItem.stage as DeploymentStageElementConfig)?.spec?.environment?.infrastructureDefinitions as any
        )
          ? RUNTIME_INPUT_VALUE
          : (stageItem.stage as DeploymentStageElementConfig)?.spec?.environment?.infrastructureDefinitions?.[0]
              ?.identifier
        const infrastructureLabel = `Infrastructure [${singleInfrastructureRef}]`

        return {
          label: `Stage [${stageItem.stage?.name}] - ${environmentLabel}`,
          value: stageItem.stage?.identifier as string,
          infraLabel: `Stage [${stageItem.stage?.name}] - ${infrastructureLabel}`
        }
      }
    })
  }
  return []
}
