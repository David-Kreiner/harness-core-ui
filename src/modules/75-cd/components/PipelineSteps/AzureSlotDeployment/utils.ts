import { defaultTo } from 'lodash-es'
import { RUNTIME_INPUT_VALUE, MultiTypeInputType } from '@harness/uicore'
import type { StageElementWrapper } from '@pipeline/utils/pipelineTypes'
import type { StageElementConfig, StageElementWrapperConfig, PipelineInfoConfig } from 'services/pipeline-ng'
import type { InfraStructureDefinitionYaml, DeploymentStageConfig, EnvironmentYamlV2 } from 'services/cd-ng'

interface EnvironmentYamlV2Runtime extends Omit<EnvironmentYamlV2, 'infrastructureDefinitions'> {
  infrastructureDefinitions?: InfraStructureDefinitionYaml[] | string
}

interface DeploymentStageConfigAzureWebName extends Omit<DeploymentStageConfig, 'environment'> {
  environment: EnvironmentYamlV2Runtime
}

export interface DeploymentStageElementConfigAzureWebName extends Omit<StageElementConfig, 'spec'> {
  spec?: DeploymentStageConfigAzureWebName
}
export type SelectedStageType = StageElementWrapper<DeploymentStageElementConfigAzureWebName>

export function getStage(stageId: string, pipeline?: PipelineInfoConfig): StageElementWrapperConfig | undefined {
  if (pipeline?.stages) {
    let responseStage: StageElementWrapperConfig | undefined = undefined
    pipeline.stages.forEach(item => {
      if (item.stage && item.stage.identifier === stageId) {
        responseStage = item
      } else if (item.parallel) {
        return item.parallel.forEach(node => {
          if (node.stage?.identifier === stageId) {
            responseStage = node
          }
        })
      }
    })
    return responseStage
  }
  return
}

export const isRuntimeEnvId = (selectedStage: SelectedStageType): boolean => {
  const envId = selectedStage?.stage?.spec?.environment?.environmentRef
  return !!(envId && envId === RUNTIME_INPUT_VALUE)
}

export const isRuntimeInfraId = (selectedStage: SelectedStageType): boolean => {
  const infraId = selectedStage?.stage?.spec?.environment?.infrastructureDefinitions
  return infraId === RUNTIME_INPUT_VALUE
}

export const isMultiEnv = (selectedStage: SelectedStageType): boolean => {
  const isEnvs = !!selectedStage?.stage?.spec?.environments || !!selectedStage?.stage?.spec?.environmentGroup
  return !!isEnvs
}

export const getEnvId = (selectedStage: SelectedStageType): string => {
  if (isMultiEnv(selectedStage || isRuntimeEnvId(selectedStage))) {
    return ''
  }

  return defaultTo(selectedStage?.stage?.spec?.environment?.environmentRef, '')
}

export const getInfraId = (selectedStage: SelectedStageType): string => {
  const envId = getEnvId(selectedStage)
  if (!envId || isMultiEnv(selectedStage)) {
    return ''
  }

  if (!isRuntimeEnvId(selectedStage) && !isRuntimeInfraId(selectedStage)) {
    const [infra] = selectedStage?.stage?.spec?.environment?.infrastructureDefinitions as InfraStructureDefinitionYaml[]
    return infra.identifier
  }

  return ''
}
/* eslint-disable */
export const getEnvIdRuntime = (stageId: string, values: any): string => {
  const stage = getStage(stageId, values) as SelectedStageType
  const envId = stage?.stage?.spec?.environment?.environmentRef
  return envId || ''
}
/* eslint-disable */
export const getInfraIdRuntime = (stageId: string, values: any): string => {
  const stage = getStage(stageId, values) as SelectedStageType

  if (stage?.stage?.spec?.environment?.infrastructureDefinitions?.[0]) {
    const [{ identifier }] = stage?.stage?.spec?.environment
      ?.infrastructureDefinitions as InfraStructureDefinitionYaml[]
    return identifier
  }
  return ''
}

export const getAllowableTypes = (selectedStage: SelectedStageType): MultiTypeInputType[] => {
  if (isMultiEnv(selectedStage)) {
    return [MultiTypeInputType.EXPRESSION]
  }

  return [MultiTypeInputType.FIXED, MultiTypeInputType.RUNTIME, MultiTypeInputType.EXPRESSION]
}
