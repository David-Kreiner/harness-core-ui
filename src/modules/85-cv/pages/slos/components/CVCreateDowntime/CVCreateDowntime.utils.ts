/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import * as Yup from 'yup'
import type { Item } from '@harness/uicore/dist/components/ThumbnailSelect/ThumbnailSelect'
import type { UseStringsReturn } from 'framework/strings'
import type { DowntimeDTO } from 'services/cv'
import { DowntimeForm, DowntimeFormFields } from './CVCreateDowntime.types'

export const getDowntimeInitialFormData = (sloDowntime?: DowntimeDTO): DowntimeForm => {
  return {
    [DowntimeFormFields.NAME]: sloDowntime?.name || '',
    [DowntimeFormFields.IDENTIFIER]: sloDowntime?.identifier || '',
    [DowntimeFormFields.TAGS]: sloDowntime?.tags,
    [DowntimeFormFields.DESCRIPTION]: sloDowntime?.description,
    [DowntimeFormFields.CATEGORY]: sloDowntime?.category || ('' as DowntimeDTO['category'])
  }
}

export const handleDowntimeSubmit = (values: any) => values

export const getDowntimeFormValidationSchema = (getString: UseStringsReturn['getString']): any => {
  return Yup.object().shape({
    [DowntimeFormFields.NAME]: Yup.string()
      .trim()
      .required(getString('cv.sloDowntime.validations.nameValidation'))
      .matches(/^[0-9a-zA-Z-_\s]+$/, getString('cv.slos.validations.specialCharacters')),
    [DowntimeFormFields.IDENTIFIER]: Yup.string().when([DowntimeFormFields.NAME], {
      is: name => name,
      then: Yup.string().trim().required(getString('validation.identifierRequired'))
    })
  })
}

export const getDowntimeCategoryOptions = (getString: UseStringsReturn['getString']): Item[] => {
  return [
    {
      label: getString('cv.sloDowntime.scheduledMaintenance'),
      value: 'ScheduledMaintenance'
    },
    {
      label: getString('deploymentText'),
      value: 'Deployment'
    },
    {
      label: getString('common.other'),
      value: 'Other'
    }
  ]
}
