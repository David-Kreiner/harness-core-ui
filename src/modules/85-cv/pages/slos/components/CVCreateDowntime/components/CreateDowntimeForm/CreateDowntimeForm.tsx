/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useCallback, useEffect, useRef, useState } from 'react'
import { useParams } from 'react-router-dom'
import { Layout, Page, Button, ButtonVariation } from '@harness/uicore'
import { useFormikContext } from 'formik'
import { useStrings } from 'framework/strings'
import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import { CVStepper } from '@cv/components/CVStepper/CVStepper'
import { CreateDowntimeFormInterface, CreateDowntimeSteps } from './CreateDowntimeForm.types'
import type { DowntimeForm } from '../../CVCreateDowntime.types'
import DowntimeName from './components/DowntimeName/DowntimeName'
import { getErrorMessageByTabId, isFormDataValid } from './CreateDowntimeForm.utils'
import { CreateDowntimePreview } from './components/CreateDowntimePreview/CreateDowntimePreview'
import css from './CreateDowntimeForm.module.scss'

export const CreateDowntimeForm = ({
  loading,
  error,
  // retryOnError,
  // handleRedirect,
  loadingSaveButton,
  runValidationOnMount
}: CreateDowntimeFormInterface): JSX.Element => {
  const { identifier } = useParams<ProjectPathProps & { identifier: string }>()
  const { getString } = useStrings()
  const formikProps = useFormikContext<DowntimeForm>()
  const isStepValid = useCallback(
    (stepId: string) => isFormDataValid(formikProps, stepId as CreateDowntimeSteps),
    [formikProps.values, formikProps.errors]
  )

  const [validateAllSteps, setValidateAllSteps] = useState<boolean | undefined>(runValidationOnMount)
  const sloDowntimePayloadRef = useRef<DowntimeForm | null>()
  const prevStepDataRef = useRef<DowntimeForm | null>()

  useEffect(() => {
    sloDowntimePayloadRef.current = formikProps.values
    prevStepDataRef.current = formikProps.values
  }, [])

  const onStepChange = (): void => {
    prevStepDataRef.current = formikProps.values
  }

  return (
    <>
      {/* add retryOnError to Page.Body afterwards */}
      <Page.Body loading={loading} error={error}>
        <>
          <CVStepper
            id="createDowntimeTabs"
            isStepValid={isStepValid}
            runValidationOnMount={validateAllSteps}
            onStepChange={onStepChange}
            stepList={[
              {
                id: CreateDowntimeSteps.DEFINE_DOWNTIME,
                title: getString('cv.sloDowntime.steps.identification'),
                panel: <DowntimeName formikProps={formikProps} identifier={identifier} />,
                errorMessage: getErrorMessageByTabId(formikProps, CreateDowntimeSteps.DEFINE_DOWNTIME),
                preview: <CreateDowntimePreview id={CreateDowntimeSteps.DEFINE_DOWNTIME} data={formikProps.values} />
              },
              {
                id: CreateDowntimeSteps.SELECT_DOWNTIME_WINDOW,
                title: getString('cv.sloDowntime.steps.identification'),
                panel: <></>,
                errorMessage: getErrorMessageByTabId(formikProps, CreateDowntimeSteps.SELECT_DOWNTIME_WINDOW),
                preview: <></>
              }
            ]}
          />
          <Page.Header
            className={css.footer}
            title={
              <Layout.Horizontal spacing="medium">
                <Button text={getString('cancel')} variation={ButtonVariation.SECONDARY} />
                <Button
                  text={getString('save')}
                  loading={loadingSaveButton}
                  variation={ButtonVariation.PRIMARY}
                  onClick={() => setValidateAllSteps(true)}
                />
              </Layout.Horizontal>
            }
          />
        </>
      </Page.Body>
    </>
  )
}
