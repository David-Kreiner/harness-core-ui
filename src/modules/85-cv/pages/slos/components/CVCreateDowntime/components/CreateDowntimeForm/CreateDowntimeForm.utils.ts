/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FormikProps } from 'formik'
import { isEmpty } from 'lodash-es'
import { DowntimeForm, DowntimeFormFields } from '../../CVCreateDowntime.types'
import { CreateDowntimeSteps } from './CreateDowntimeForm.types'

export const isFormDataValid = (
  formikProps: FormikProps<DowntimeForm>,
  selectedTabId: CreateDowntimeSteps
): boolean => {
  switch (selectedTabId) {
    case CreateDowntimeSteps.DEFINE_DOWNTIME:
      return validateDefineDowntimeSection(formikProps)
    default:
      return false
  }
}

export const validateDefineDowntimeSection = (formikProps: FormikProps<DowntimeForm>): boolean => {
  formikProps.setFieldTouched(DowntimeFormFields.NAME, true)
  formikProps.setFieldTouched(DowntimeFormFields.IDENTIFIER, true)
  formikProps.setFieldTouched(DowntimeFormFields.CATEGORY, true)

  const isNameValid = /^[0-9a-zA-Z-_\s]+$/.test(formikProps.values['name'])
  const { name, identifier, category } = formikProps.values
  if (!name || isEmpty(identifier) || isEmpty(category) || !isNameValid) {
    return false
  }
  return true
}

export const getErrorMessageByTabId = (
  formikProps: FormikProps<DowntimeForm>,
  selectedTabId: CreateDowntimeSteps
): string[] => {
  switch (selectedTabId) {
    case CreateDowntimeSteps.DEFINE_DOWNTIME:
      return errorDefineSLOSection(formikProps.errors)
    default:
      return []
  }
}

const errorDefineSLOSection = (errors: FormikProps<DowntimeForm>['errors']) => {
  const { name, identifier, category } = errors
  return [name, identifier, category].filter(item => Boolean(item)) as string[]
}
