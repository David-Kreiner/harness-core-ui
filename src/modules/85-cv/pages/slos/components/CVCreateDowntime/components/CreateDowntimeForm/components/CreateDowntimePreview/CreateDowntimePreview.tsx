/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Layout, Text } from '@harness/uicore'
import { Color } from '@harness/design-system'
import { useStrings } from 'framework/strings'
import { CreateDowntimeSteps } from '../../CreateDowntimeForm.types'
import type { CreateDowntimePreviewProps, LabelValueProps } from './CreateDowntimePreview.types'

export const LabelAndValue = ({ label, value }: LabelValueProps): JSX.Element => {
  return (
    <Layout.Horizontal spacing="medium">
      <Text title={label} font={{ weight: 'semi-bold' }} color={Color.GREY_1000} width={100}>
        {label}
      </Text>
      <Text font={{ weight: 'light' }} color={Color.GREY_1000}>
        {value}
      </Text>
    </Layout.Horizontal>
  )
}

export const CreateDowntimePreview = ({ id, data }: CreateDowntimePreviewProps): JSX.Element => {
  const { getString } = useStrings()
  switch (id) {
    case CreateDowntimeSteps.DEFINE_DOWNTIME:
      return (
        <Layout.Vertical spacing="medium">
          <LabelAndValue label={getString('cv.sloDowntime.downtimeName')} value={data.name} />
          {data.description && <LabelAndValue label={getString('description')} value={data.description} />}
          <LabelAndValue label={getString('cv.slos.userJourney')} value={data.category} />
        </Layout.Vertical>
      )
    default:
      return <></>
  }
}
