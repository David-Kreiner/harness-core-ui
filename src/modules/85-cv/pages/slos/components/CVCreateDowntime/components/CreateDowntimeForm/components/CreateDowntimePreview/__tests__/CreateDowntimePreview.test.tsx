/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { getDowntimeInitialFormData } from '@cv/pages/slos/components/CVCreateDowntime/CVCreateDowntime.utils'
import { downtimeResponse } from '@cv/pages/slos/components/CVCreateDowntime/__tests__/CVCreateDowntime.mock'
import type { DowntimeDTO } from 'services/cv'
import { CreateDowntimeSteps } from '../../../CreateDowntimeForm.types'
import { CreateDowntimePreview } from '../CreateDowntimePreview'

describe('CreateDowntimePreview', () => {
  test('should render CreateDowntimePreview with empty value', async () => {
    const { container } = render(
      <TestWrapper>
        <CreateDowntimePreview id={'' as any} data={{} as any} />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('should render CreateDowntimePreview with init value', async () => {
    const { container } = render(
      <TestWrapper>
        <CreateDowntimePreview id={CreateDowntimeSteps.DEFINE_DOWNTIME} data={getDowntimeInitialFormData()} />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('should render CreateDowntimePreview with correct values', async () => {
    const { container } = render(
      <TestWrapper>
        <CreateDowntimePreview
          id={CreateDowntimeSteps.DEFINE_DOWNTIME}
          data={getDowntimeInitialFormData(downtimeResponse.resource.downtime as DowntimeDTO)}
        />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })
})
