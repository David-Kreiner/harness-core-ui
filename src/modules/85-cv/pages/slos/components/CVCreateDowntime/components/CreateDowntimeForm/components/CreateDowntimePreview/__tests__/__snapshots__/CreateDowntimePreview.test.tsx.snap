// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateDowntimePreview should render CreateDowntimePreview with correct values 1`] = `
<div>
  <div
    class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
  >
    <div
      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey1000"
        style="width: 100px;"
        title="cv.sloDowntime.downtimeName"
      >
        cv.sloDowntime.downtimeName
      </p>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-weight-light StyledProps--color StyledProps--color-grey1000"
      >
        SLO Downtime
      </p>
    </div>
    <div
      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey1000"
        style="width: 100px;"
        title="description"
      >
        description
      </p>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-weight-light StyledProps--color StyledProps--color-grey1000"
      >
        Weekly downtime
      </p>
    </div>
    <div
      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey1000"
        style="width: 100px;"
        title="cv.slos.userJourney"
      >
        cv.slos.userJourney
      </p>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-weight-light StyledProps--color StyledProps--color-grey1000"
      >
        Deployment
      </p>
    </div>
  </div>
</div>
`;

exports[`CreateDowntimePreview should render CreateDowntimePreview with empty value 1`] = `<div />`;

exports[`CreateDowntimePreview should render CreateDowntimePreview with init value 1`] = `
<div>
  <div
    class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
  >
    <div
      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey1000"
        style="width: 100px;"
        title="cv.sloDowntime.downtimeName"
      >
        cv.sloDowntime.downtimeName
      </p>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-weight-light StyledProps--color StyledProps--color-grey1000"
      >
        
      </p>
    </div>
    <div
      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey1000"
        style="width: 100px;"
        title="cv.slos.userJourney"
      >
        cv.slos.userJourney
      </p>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-weight-light StyledProps--color StyledProps--color-grey1000"
      >
        
      </p>
    </div>
  </div>
</div>
`;
